zerver/lib/avatar.py

```diff
@@ -117,3 +117,18 @@ def absolute_avatar_url(user_profile: UserProfile) -> str:
     # avatar_url can return None if client_gravatar=True, however here we use the default value of False
     assert avatar is not None
     return urllib.parse.urljoin(user_profile.realm.uri, avatar)
+
+def is_avatar_new(ldap_avatar: bytes, user_profile: UserProfile) -> bool:
+    new_avatar_hash = user_avatar_content_hash(ldap_avatar)
+
+    # If an avatar hash already exists,
+    # then check if it is the same as the new avatar's hash.
+    if user_profile.avatar_hash:
+        if user_profile.avatar_hash != new_avatar_hash:
+            user_profile.avatar_hash = new_avatar_hash
+            return True
+    else:
+        user_profile.avatar_hash = new_avatar_hash
+        return True
```

Comment:
You need `user_profile.save(update_fields=["avatar_hash"])` to actually make the change in the database.  (Otherwise, you just adjusted a Python object and that's it)

Reply:
(That said, I would put the `.save()` type code in the caller, since changing this value makes the function not match its name, and we really only want to update the hash once the avatar is finished uploading)

Reply:
Made the change to update `avatar_hash` once the update is done.